src/Http/Http.Features/src/IResponseCookies.cs

```diff
@@ -23,6 +25,13 @@ public interface IResponseCookies
         /// <param name="options"><see cref="CookieOptions"/> included in the new cookie setting.</param>
         void Append(string key, string value, CookieOptions options);
 
+        /// <summary>
+        /// Add elements of specified dictionary as cookies.
+        /// </summary>
+        /// <param name="keyValuePairs">Key value pair collections whose elements will be added as cookies.</param>
+        /// <param name="options"><see cref="CookieOptions"/> included in new cookie settings.</param>
+        void Append(IDictionary<string, string> keyValuePairs, CookieOptions options);
```

Comment:
Does this need to be an `IDictionary<string, string>` or could it be less specific like IEnumerable<KeyValuePair<string, string>>`

Reply:
Yep, you are right `IEnumerable<KeyValuePair<string, string>>` is even more performant that `IDictionary` implementation, I am changing it